src/core/plugin/luapi_application.h

```diff
@@ -2280,6 +2280,35 @@ static int applib_scrollToPos(lua_State* L) {
     return 0;
 }
 
+/**
+ * Obtains the label of the specified page in the pdf background.
+ *
+ * @param page integer
+ * @return string|nil Returns the label on success and (nil, message) if page number is out of range.
+ * @return string
+ *
+ * Example 1: local label = app.getPageLabel(10)
+ ' obtains the page label of page 10 in the background pdf
+ **/
+static int applib_getPageLabel(lua_State* L) {
+    const Plugin* plugin = Plugin::getPluginFromLua(L);
+    const Control* control = plugin->getControl();
+    const XojPdfDocument& doc = control->getDocument()->getPdfDocument();
+
+    const size_t pagenr = static_cast<size_t>(luaL_checkinteger(L, 1)) - 1;
+    const size_t last = doc.getPageCount() - 1;
+
+    if (pagenr > last) {
+        lua_pushnil(L);
+        lua_pushfstring(L, "page nr %d is out of range", pagenr);
+        return 2;
+    }
+
+    std::string label = doc.getPage(pagenr)->getPageLabel();
```

Comment:
Strictly speaking we should lock the document during access here, right? There are other places in the Lua API code though, where we also don't do it.

Reply:
Yea we had this on [gitter once](https://matrix.to/#/!mIWOZbAzBNmtxiyIii:gitter.im/$-z71mhwupQ1YqkAnGLlTrNYmNvXegvq0m7Wq6rHu0OQ?via=gitter.im&via=matrix.org&via=beeper.com). Tbh I didn't check what happened afterwards, but my POV on this was we'd try to solve this more general than inserting locks in the `luapi_application.h`. But you might argue that this leads to unstable behavior and we should do our best in the luapi to do the locking (and only after we fixed this more generally we should remove the lock statements).
Coming from the *avoid crashes* side, I think I'd be in favor to add the locks in the luapi. Would be a new PR in my opinion though.

Reply:
Tbh, I'm not sure which functions need locking. For now I just said every set referring to the document (not the control etc) should acquire a lock before executing the action. (Not sure if it's correct to spare the getters)
First suggestion at https://github.com/atticus-sullivan/xournalpp/tree/luapi-locking

Any comments? (I made a draft PR here https://github.com/xournalpp/xournalpp/pull/5888)